cg/meta/deliver_ticket.py

```diff
@@ -0,0 +1,135 @@
+"""Module for deliver and rsync customer inbox on hasta to customer inbox on caesar"""
+import logging
+import os
+import re
+import subprocess
+from pathlib import Path
+import shutil
+from typing import List
+
+from cg.exc import CgError
+from cg.meta.meta import MetaAPI
+from cg.models.cg_config import CGConfig
+from cg.store import models
+
+
+LOG = logging.getLogger(__name__)
+PREFIX_TO_CONCATENATE = ["MWG", "MWL", "MWM", "MWR", "MWX"]
+
+
+class DeliverTicketAPI(MetaAPI):
+    def __init__(self, config: CGConfig):
+        super().__init__(config)
+        self.delivery_path: str = config.delivery_path
+
+    def get_all_cases_from_ticket(self, ticket_id: int) -> list:
+        cases: List[models.Family] = self.status_db.get_cases_from_ticket(ticket_id=ticket_id).all()
+        return cases
+
+    def get_inbox_path(self, ticket_id: int) -> str:
+        cases = self.get_all_cases_from_ticket(ticket_id=ticket_id)
+        if not cases:
+            LOG.warning("Could not find any cases for ticket_id %s", ticket_id)
+            raise CgError()
+
+        customer_id: str = cases[0].customer.internal_id
+        customer_inbox: str = (
+            Path(self.delivery_path, customer_id, "inbox", str(ticket_id)).as_posix() + "/"
+        )
+        return customer_inbox
+
+    def check_if_upload_is_needed(self, ticket_id: int) -> bool:
+        is_upload_needed = True
+        customer_inbox = self.get_inbox_path(ticket_id=ticket_id)
+        LOG.info("Checking if path exist: %s", customer_inbox)
+        if os.path.exists(customer_inbox):
+            LOG.info("Could find path: %s", customer_inbox)
+            is_upload_needed = False
+        else:
+            LOG.info("Could not find path: %s", customer_inbox)
+        return is_upload_needed
+
+    def generate_date_tag(self, ticket_id: int) -> str:
+        cases = self.get_all_cases_from_ticket(ticket_id=ticket_id)
+        date = str(cases[0].ordered_at)
+        split_date = date.split(" ")
+        return split_date[0]
+
+    def concatenate(self, ticket_id: int) -> None:
```

Comment:
i think it is much easier to write new files to the cust inbox (earlier on, via deliver analysis) than hardlinking, concatenating, and then deleting original files by looping through delivery folder. Saves time on read/write operations and prolongs hard disk life!